custom_image_utils/args_inferer.py

```diff
@@ -128,12 +128,22 @@ def _get_dataproc_image_path_by_version(version):
   """Get Dataproc base image name from version."""
   # version regex already checked in arg parser
   parsed_version = version.split(".")
-  filter_arg = "--filter=labels.goog-dataproc-version=\'{}-{}-{}\'".format(
-      parsed_version[0], parsed_version[1], parsed_version[2])
-  command = [
-      "gcloud", "compute", "images", "list", "--project", "cloud-dataproc",
-      filter_arg, "--format=csv[no-heading=true](name,status)"
-  ]
+  if len(parsed_version) == 2:
```

Comment:
IIUC, it simply gets the latest image based on the minor version, but never resolves the minor version to a subminor version and `goog-dataproc-version=<minor-version>` label will be added to the custom image. This is problematic, because Dataproc control plane needs to know the subminor version to determine the metadata for the image (e.g., what optional components are available, what properties are supported, etc).

I think here it needs to no only resolve the image, but also resolve the subminor version, and replace the minor version with the subminor version forward.